frontend/src/components/nlp/components/NlpSample.tsx

```diff
@@ -343,13 +368,25 @@ export default function NlpSample() {
               EntityType.NLP_SAMPLE_ENTITY,
               PermissionAction.CREATE,
             ) ? (
-              <Button
-                variant="contained"
-                onClick={() => importDialogCtl.openDialog()}
-                startIcon={<UploadIcon />}
-              >
-                {t("button.import")}
-              </Button>
+              <>
+                <Button
+                  htmlFor="importFile"
+                  variant="contained"
+                  component="label"
+                  startIcon={<UploadIcon />}
+                  endIcon={isLoading ? <CircularProgress size="1rem" /> : null}
+                  disabled={isLoading}
+                >
+                  {t("button.import")}
+                </Button>
+                <Input
+                  id="importFile"
+                  type="file"
+                  value="" // to trigger an automatic reset to allow the same file to be selected multiple times
+                  sx={{ display: "none" }}
+                  onChange={handleImportChange}
+                />
+              </>
```

Comment:
Let's create a dedicated component, that should, given a mime type as a prop do some checks.